src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -56,25 +57,59 @@ public async Task<ImmutableArray<CodeAction>> AddConstructorParametersFromMember
 
         private IEnumerable<CodeAction> CreateCodeActions(Document document, State state)
         {
-            var parameters = state.ConstructorToAddTo.Parameters;
-            if (parameters.Length == 0 ||
-                (parameters.Length > 0 && !parameters.Last().IsOptional))
+            var result = new ArrayBuilder<CodeAction>();
+            var containingType = state.ContainingType;
+            if (state.ConstructorCandidates.Count() <= 1)
             {
-                // return a code action to add required parameters
-                yield return new AddConstructorParametersCodeAction(this, document, state, state.MissingParameters);
+                // There will be at most 2 suggested code actions, so no need to use sub menus
+                var constructorCandidate = state.ConstructorCandidates[0];
```

Comment:
1. the check above says `<=`, which means we culd have 0, so this is unsafe to do.

Reply:
The check now uses `==`